File: golang/internal/multiapp/context.go
package multiapp

import (
	"rocket-backend/internal/admin"
	"rocket-backend/internal/auth"
	"rocket-backend/internal/engine"
	"rocket-backend/internal/metadata"
	"rocket-backend/internal/store"
)

// AppContext holds all per-app resources: database pool, metadata cache, and pre-built handlers.
type AppContext struct {
	Name      string
	DBName    string
	JWTSecret string

	Store    *store.Store
	Registry *metadata.Registry
	Migrator *store.Migrator

	EngineHandler   *engine.Handler
	AdminHandler    *admin.Handler
	AuthHandler     *auth.AuthHandler
	WorkflowHandler *engine.WorkflowHandler
}

// BuildHandlers creates all handler instances for this app context.
func (ac *AppContext) BuildHandlers() {
	ac.Migrator = store.NewMigrator(ac.Store)
	ac.EngineHandler = engine.NewHandler(ac.Store, ac.Registry)
	ac.AdminHandler = admin.NewHandler(ac.Store, ac.Registry, ac.Migrator)
	ac.AuthHandler = auth.NewAuthHandler(ac.Store, ac.JWTSecret)
	ac.WorkflowHandler = engine.NewWorkflowHandler(ac.Store, ac.Registry)
}

// AppInfo is a summary of an app returned by List.
type AppInfo struct {
	Name        string `json:"name"`
	DisplayName string `json:"display_name"`
	DBName      string `json:"db_name"`
	Status      string `json:"status"`
	CreatedAt   any    `json:"created_at"`
	UpdatedAt   any    `json:"updated_at"`
}

File: expressjs/src/multiapp/context.ts
import type { Store } from "../store/postgres.js";
import type { Registry } from "../metadata/registry.js";
import type { Migrator } from "../store/migrator.js";
import type { Handler } from "../engine/handler.js";
import type { AdminHandler } from "../admin/handler.js";
import type { AuthHandler } from "../auth/handler.js";
import type { WorkflowHandler } from "../engine/workflow-handler.js";

export interface AppContext {
  name: string;
  dbName: string;
  jwtSecret: string;
  store: Store;
  registry: Registry;
  migrator: Migrator;
  engineHandler: Handler;
  adminHandler: AdminHandler;
  authHandler: AuthHandler;
  workflowHandler: WorkflowHandler;
}

export interface AppInfo {
  name: string;
  display_name: string;
  db_name: string;
  status: string;
  created_at: any;
  updated_at: any;
}
